### PR TITLE
Feature/cdm line item

### PR DESCRIPTION
**Pull request by fivetran-catfritz**
## PR Overview
**This PR will address the following Issue/Feature:**
- Quarterly project

**This PR will result in the following new package version:**

- v0.3.0 for new model

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

> ## 🚨 Breaking Changes 🚨
> - The following columns were added to model `stg_recharge__address`:
>   - country
>   - payment_method_id
>   - ❗Note: If you have already added any of these fields as passthrough columns to the `recharge__address_passthrough_columns` var, you will need to remove or alias these fields from the var to avoid duplicate column errors.
> 
> ## Features
> - Added staging model `stg_recharge__checkout`. See [this doc](https://fivetran.github.io/dbt_recharge_source/#!/model/model.recharge_source.stg_recharge__checkout) for the fields added and their definitions.
>   - This model is disabled by default but can be enabled by setting variable `recharge__checkout_enabled` to true in your `dbt_project.yml` file. See the [Enable/disable models and sources section](https://github.com/fivetran/dbt_recharge_source/blob/main/README.md#step-4-enable-disable-models-and-sources) of the README for more information.
>   - This model can also be passed additional columns beyond the predefined columns by using the variable `recharge__checkout_passthrough_columns`. See the [Passing Through Additional Columns](https://github.com/fivetran/dbt_recharge_source/blob/main/README.md#passing-through-additional-columns) section of the README for more information on how to set this variable.
> 
> - Added the following columns to model `stg_recharge__customer`. See [this doc](https://fivetran.github.io/dbt_recharge_source/#!/model/model.recharge_source.stg_recharge__customer) for field definitions.
>   - `billing_first_name`
>   - `billing_last_name`
>   - `billing_company`
>   - `billing_city`
>   - `billing_country`
> 
> ## Under the hood
> - Updated `stg_recharge__subscription_history` to coalesce `id` and `subscription_id` as the `subscription_id` since either version can be present in the source.

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [ ] ~dbt run (if incremental models are present) && dbt test~

Before marking this PR as "ready for review" the following have been applied:
- [ ] ~The appropriate issue has been linked, tagged, and properly assigned~
- [x] All necessary documentation and version upgrades have been applied
- [x] docs were regenerated (unless this PR does not include any code or yml updates)
- [x] BuildKite integration tests are passing
- [x] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

- See internal ticket for schema sample.

### If you had to summarize this PR in an emoji, which would it be?
 
:dancer:

### CODE REVIEW

File: macros/get_address_columns.sql [hunk 1]
```diff
@@ -11,12 +11,14 @@
...
     {"name": "country_code", "datatype": dbt.type_string()},
+    {"name": "country", "datatype": dbt.type_string()},
     {"name": "zip", "datatype": dbt.type_string()},
     {"name": "company", "datatype": dbt.type_string()},
     {"name": "phone", "datatype": dbt.type_string()},
     {"name": "created_at", "datatype": dbt.type_timestamp()},
     {"name": "updated_at", "datatype": dbt.type_timestamp()},
-    {"name": "_fivetran_deleted", "datatype": dbt.type_boolean()}
+    {"name": "_fivetran_deleted", "datatype": dbt.type_boolean()},
+    {"name": "payment_method_id", "datatype": dbt.type_string()}
```
Comment (fivetran-joemarkiewicz) [inline macros/get_address_columns.sql:21]: A customer _could_ have had `country` and/or `payment_method_id` as a passthrough column for this model. Let's be sure to list this as a potential breaking change once the CHANGELOG is generated.

File: models/stg_recharge__checkout.sql [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        order_attributes,
+        phone,
+        requires_shipping,
+        subtotal_price,
+        taxes_included,
+        total_price,
+        total_tax,
+        cast(updated_at as {{ dbt.type_timestamp() }}) as checkout_updated_at
+
+    {{ fivetran_utils.fill_pass_through_columns('recharge__checkout_passthrough_columns') }}
```
Comment (fivetran-joemarkiewicz) [inline models/stg_recharge__checkout.sql:44]: Be sure to include documentation for this new passthrough column once the docs are ready to be updated.

File: models/tmp/stg_recharge__checkout.sql [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{{ config(enabled=var('recharge__using_checkout', false)) }}
```
Comment (fivetran-joemarkiewicz) [inline models/tmp/stg_recharge__checkout.sql]: Be sure to document this variable once the docs are ready to be created

File: integration_tests/seeds/checkout_data.csv [hunk 4]
```diff
@@ -0,0 +1,2 @@
+token, charge_id
```
Comment (fivetran-joemarkiewicz) [inline integration_tests/seeds/checkout_data.csv]: Reminder to add more seed fields before release

File: models/stg_recharge.yml [hunk 5]
```diff
@@ -280,6 +280,9 @@ models:
       - name: title
         description: The title/name of the taxing jurisdiction.
 
+  - name: stg_recharge__checkout
```
Comment (fivetran-joemarkiewicz) [inline models/stg_recharge.yml:283]: Reminder to update the docs when ready

File: models/src_recharge.yml [hunk 6]
```diff
@@ -267,6 +267,12 @@ sources:
           - name: _fivetran_synced
             description: "{{ doc('_fivetran_synced') }}"
 
+      - name: checkout
```
Comment (fivetran-joemarkiewicz) [inline models/src_recharge.yml:270]: Reminder to update docs when ready

File: models/stg_recharge__address.sql [hunk 7]
```diff
@@ -32,8 +32,10 @@ final as (
         province,
         zip,
         country_code,
+        country,
         company,
-        phone
+        phone,
+        payment_method_id
```
Comment (fivetran-joemarkiewicz) [inline models/stg_recharge__address.sql:38]: We technically will want to make this a breaking change and call it out in the CHANGELOG. These fields _could_ have been used in the passthrough column variable and could result in a breaking run if they are still included.

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-catfritz no major code changes needed in this PR! The only changes required before approving are including the necessary documentation and CHANGELOG updates.

File: models/stg_recharge__checkout.sql (hunk 2 above)
Comment (fivetran-catfritz) [reply models/stg_recharge__checkout.sql:44]: Added.

File: models/stg_recharge__address.sql (hunk 7 above)
Comment (fivetran-catfritz) [reply models/stg_recharge__address.sql:38]: Updated!

File: models/stg_recharge.yml (hunk 5 above)
Comment (fivetran-catfritz) [reply models/stg_recharge.yml:283]: Updated!

File: integration_tests/seeds/checkout_data.csv (hunk 4 above)
Comment (fivetran-catfritz) [reply integration_tests/seeds/checkout_data.csv]: Updated!

File: macros/get_address_columns.sql (hunk 1 above)
Comment (fivetran-catfritz) [reply macros/get_address_columns.sql:21]: Updated!

File: models/src_recharge.yml (hunk 6 above)
Comment (fivetran-catfritz) [reply models/src_recharge.yml:270]: Updated!

File: models/tmp/stg_recharge__checkout.sql (hunk 3 above)
Comment (fivetran-catfritz) [reply models/tmp/stg_recharge__checkout.sql]: Added.

File: README.md [hunk 8]
```diff
@@ -53,17 +53,18 @@ vars:
...
 
-## Step 4: Disable models for non-existent sources
+## Step 4: Enable/disable models and sources
 Your Recharge connector may not sync every table that this package expects. If you do not have the `ONE_TIME_PRODUCT` and/or `CHARGE_TAX_LINE` tables synced, add the corresponding variable(s) to your root `dbt_project.yml` file to disable these sources:
 
 ```yml
 vars:
   recharge__one_time_product_enabled: false # Disables if you do not have the ONE_TIME_PRODUCT table. Default is True.
   recharge__charge_tax_line_enabled: false # Disables if you do not have the CHARGE_TAX_LINE table. Default is True.
+  recharge__checkout_enabled: false # Disables if you do not have the CHECKOUT table. Default is True.
```
Comment (fivetran-joemarkiewicz) [inline README.md]: Isn't the default `false`?

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-catfritz Looks great! Just one documentation question to address before approval.

File: README.md (hunk 8 above)
Comment (fivetran-catfritz) [reply README.md]: Yes, updated!

**Review by fivetran-catfritz — COMMENTED** [review]
This is now updated!

**Review by fivetran-joemarkiewicz — APPROVED** [review]
LGTM!

File: models/src_recharge.yml [hunk 9]
```diff
@@ -267,6 +267,51 @@ sources:
...
+        identifier: "{{ var('recharge_checkout_identifier','checkout') }}"
+        description: Table of checkout information associated to the corresponding charges.
+        config:
+          enabled: "{{ var('recharge__checkout_enabled', false) }}"
+        columns:
+          - name: token
```
Comment (fivetran-reneeli) [inline models/src_recharge.yml:276]: Do we want to add a unique test here?

**Review by fivetran-reneeli — APPROVED** [review]
looks good!

**Pull request merged**